#!/usr/bin/env python3
PACKAGE = "robot_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x_position_2",   double_t,   0, "x value for 2 position", 0.479  , 0.2 , 0.8)
gen.add("x_position_1",   double_t,   0, "x value for 1 position", 0.4025  , 0.2 , 0.8)
gen.add("x_position_0",   double_t,   0, "x value for 0 position", 0.33  , 0.2 , 0.8)

gen.add("height_drop",   double_t,   0, "height from which the disk is dropped", 0.05  , 0.0 , 0.2)
gen.add("disk_height",  double_t,   0, "height of the disk", 0.009  , 0.002 , 0.020)

gen.add("using_computer_vision",  bool_t,   0, "using detected or hardcoded rods positions", False)

gen.add("rods_width_threshold",  double_t,   0, 
    "if a object smaller than that is grabbed, it is considered a rod", 0.01  , 0.002 , 0.02)



exit(gen.generate(PACKAGE, "hanoi", "Hanoi"))